src/test/java/com/mongodb/kafka/connect/util/FlexibleDateTimeParserTest.java

```diff
@@ -0,0 +1,118 @@
+        () ->
+            assertEquals(
+                3600000,
+                DEFAULT_DATE_TIME_PARSER.toEpochMilli("1970-01-01 02:00:00 +01:00[Europe/Paris]")),
+        () ->
+            assertEquals(
+                3600000,
+                DEFAULT_DATE_TIME_PARSER.toEpochMilli("1970-01-01T02:00:00+01:00[Europe/Paris]")),
+        () ->
+            assertEquals(
+                3600000, DEFAULT_DATE_TIME_PARSER.toEpochMilli("1970-01-01T00:00:00.000001-0100")),
+        () ->
+            assertEquals(
+                3600001, DEFAULT_DATE_TIME_PARSER.toEpochMilli("1970-01-01T00:00:00.001-0100")),
+        () ->
+            assertEquals(
+                3600000, DEFAULT_DATE_TIME_PARSER.toEpochMilli("1970-01-01T00:00:00-0100")),
+        () ->
+            assertEquals(
+                3600000, DEFAULT_DATE_TIME_PARSER.toEpochMilli("1970-01-01T01:00:00.000001Z")),
+        () ->
+            assertEquals(
+                3600001, DEFAULT_DATE_TIME_PARSER.toEpochMilli("1970-01-01T01:00:00.001Z")),
+        () ->
+            assertEquals(3600001, DEFAULT_DATE_TIME_PARSER.toEpochMilli("1970-01-0101:00:00.001")),
+        () -> assertEquals(3600000, DEFAULT_DATE_TIME_PARSER.toEpochMilli("1970-01-0101:00:00")),
+        () ->
+            assertEquals(
+                3600000, DEFAULT_DATE_TIME_PARSER.toEpochMilli("1970-01-01 00:00:00 -0100")),
+        () -> assertEquals(86400000, DEFAULT_DATE_TIME_PARSER.toEpochMilli("1970-01-02")));
+  }
+
+  @Test
+  @DisplayName("test errors parsing invalid strings")
+  void testErrorsParsingInvalidStrings() {
+    assertAll(
+        () ->
+            assertThrows(
+                DateTimeParseException.class,
+                () -> DEFAULT_DATE_TIME_PARSER.toEpochMilli("1970"),
+                "Missing month and day"),
+        () ->
+            assertThrows(
+                DateTimeParseException.class,
+                () -> DEFAULT_DATE_TIME_PARSER.toEpochMilli("1970-01"),
+                "Missing day"),
+        () ->
+            assertThrows(
+                DateTimeParseException.class,
+                () -> DEFAULT_DATE_TIME_PARSER.toEpochMilli("1970-01-01T00:01"),
+                "Missing Seconds"),
+        () ->
+            assertThrows(
+                DateTimeParseException.class,
+                () -> DEFAULT_DATE_TIME_PARSER.toEpochMilli("epoch"),
+                "Normal string"),
+        () ->
+            assertThrows(
+                DateTimeParseException.class,
+                () -> DEFAULT_DATE_TIME_PARSER.toEpochMilli("1970-01-01 -0100"),
+                "Offset with no time data"),
+        () ->
+            assertThrows(
+                DateTimeParseException.class,
+                () ->
+                    DEFAULT_DATE_TIME_PARSER.toEpochMilli("1970-01-01T02:00:00+01:00[Asia/Paris]"),
+                "Invalid Zone"));
+  }
+
+  @Test
+  @DisplayName("Alternative date time formats")
+  void testAlternativeDateTimeTormats() {
```

Comment:
We discovered that the behavior of this test depends on the default locale, e.g., if the corresponding language tag is `en`, then the test passes, but if it is `en-CA`, then the test fails. As a result, Ross proposed to specify the default locale explicitly and allow users to override the explicit default.

Reply:
Great catch as that would have been found by a user and required a patch release :+1: